src/meds_torch/data/components/multiwindow_pytorch_dataset.py

```diff
@@ -14,6 +14,131 @@
 from meds_torch.data.components.pytorch_dataset import PytorchDataset
 
 
+def fuse_window_data(windows_data: dict, windows_to_fuse: list[str], fused_window_name: str) -> dict:
+    """Fuse multiple windows into a single window, tracking the lengths of original windows.
+
+    Args:
+        windows_data (dict): Dictionary containing data from multiple windows to be fused
+        windows_to_fuse (list[str]): List of window names to fuse in specified order
+        fused_window_name (str): Name for the resulting fused window
+
+    Returns:
+        dict: Fused window data with length tracking information
+
+    Raises:
+        ValueError: If fusion configuration is invalid or data type is unsupported
+
+    Example:
+    >>> import torch
+    >>> # Create mock data with different types
+    >>> windows_data = {
+    ...     "pre": {
+    ...         # 1D tensor
+    ...         "static_values": torch.tensor([1.0, 2.0]),
+    ...         # 2D tensor (batch_size=2, seq_len=3, features=2)
+    ...         "embeddings": torch.ones(2, 3, 2),
+    ...         # List
+    ...         "codes": ["A", "B", "C"],
+    ...         # Scalar
+    ...         "scalar": 42
+    ...     },
+    ...     "post": {
+    ...         "static_values": torch.tensor([3.0, 4.0]),
+    ...         "embeddings": torch.ones(2, 2, 2) * 2,
+    ...         "codes": ["D", "E"],
+    ...         "scalar": 43
+    ...     }
+    ... }
+    >>> # Fuse windows
+    >>> fused = fuse_window_data(
+    ...     windows_data,
+    ...     windows_to_fuse=["pre", "post"],
+    ...     fused_window_name="fused"
+    ... )
+    >>> # Check lengths tracking
+    >>> fused["LENGTHS//static_values"]  # Two windows with lengths 2 and 2
+    [2, 2]
+    >>> fused["LENGTHS//embeddings"]  # Two windows with sequence lengths 3 and 2
+    [3, 2]
+    >>> fused["LENGTHS//codes"]  # List lengths: 3 and 2
+    [3, 2]
+    >>> fused["LENGTHS//scalar"]  # Scalar values: count as length 1 each
+    [1]
+    >>> # Check concatenated values
+    >>> torch.equal(fused["static_values"],
+    ...            torch.cat([torch.tensor([1.0, 2.0]),
+    ...                      torch.tensor([3.0, 4.0])]))
+    True
+    >>> # Check 2D tensor concatenation along sequence dimension
+    >>> torch.equal(fused["embeddings"],
+    ...            torch.cat([torch.ones(2, 3, 2),
+    ...                      torch.ones(2, 2, 2) * 2], dim=1))
+    True
+    >>> fused["codes"]  # Lists are concatenated
+    ['A', 'B', 'C', 'D', 'E']
+    >>> fused["scalar"]  # First scalar value is kept
+    42
+    >>> # Test error handling
+    >>> fuse_window_data(windows_data, ["nonexistent"], "fused")  # doctest: +IGNORE_EXCEPTION_DETAIL
```

Comment:
You can eliminate this comment and just add a `...` in a line between the traceback and error lines.